### ISSUE TITLE
Update to Ant 1.10.13

### ISSUE THREAD

**Issue by SarikaSinha** [eclipse-platform/eclipse.platform#311]
https://github.com/apache/ant/blob/master/WHATSNEW#L1

Source and binary distributions are available for download from the
Apache Ant download site:

https://ant.apache.org/bindownload.cgi

Reference to Update for Ant 1.1.0.12 https://bugs.eclipse.org/bugs/show_bug.cgi?id=576370

Changes from Ant 1.10.12 TO Ant 1.10.13
=======================================

Changes that could break older environments:
-------------------------------------------

 * <get> has a new attribute authenticateOnRedirect that can be used to
   prevent Ant from sending the configured credentials when following a
   redirect. It is false by default, which means builds that rely on
   credentials being used on the redirected URI may break.
   Github Pull Request #173

Fixed bugs:
-----------

 * the PropertyEnumerator change introduced in 1.10.9 proved to be not
   fully backwards compatible when combined with certain custom
   PropertyHelper implementations - for example when using AntXtras.
   Bugzilla Report 65799

 * legacy-xml reporter of the junitlauncher task now escapes ]]> when writing CDATA.
   Bugzilla Report 65833

 * <scp> may leak connections when trying to preserve the last modified
   timestamps of files transferred recursively from a server.
   Bugzilla Report 66001

 * tstamp task would in certain cases parse the SOURCE_DATE_EPOCH environment variable
   value to an incorrect date. This has now been fixed.
   Github Pull Request #186

 * fetch.xml didn't set up non-default repositories propery and thus
   failed to download JAI.
   Github Pull Request #191

 * When building and installing Ant distribution from source, the build script
   would change permissions on unrelated files in the destination directory.
   This is now fixed and such unrelated files in the destination directory
   will be left untouched.
   Bugzilla Report 66164

 * parsing tar entries with multiple NUL bytes in their name would
   include garbage bytes as the name included all bytes up to the last
   NUL rather than the first.
   Github Pull Request #194

 * loadresource might log warnings even though quiet was set to true
   Bugzilla Report 65647

 * javac task would add paths constructs containing wildcards to the
   internally created argument file where wildcards are not allowed
   Bugzilla Report 65621

Other changes:
--------------

 * added an implementation of the MIME Mail sender based on the
   repackaged Jakarta Mail package rather than javax Mail.
   Github Pull Request #161

 * The "listener" element in the junitlauncher task now supports
   an "extension" attribute to control the filename extension
   of the generated output file from the listener.
   Github Pull Request #168

 * <ftp> now supports FTPs.
   Github Pull Request #170

 * DirectoryScanner avoids listing directory contents when it known it
   will never use the information retrieved. This may improve
   performance in some special cases.
   Bugzilla Report 66048

 * <manifest> will now create the parent directory of the manifestFile
   attribute if it doesn't exist.
   Bugzilla Report 66231

 * org.apache.tools.ant.BuildLogger now has a new method getMessageOutputLevel()
   which returns the currently set message output level.

**Comment by akurtakov** [comment]
@merks Do you think you can help with this on Orbit side and I'll take the platform side after that?

**Comment by merks** [comment]
@akurtakov I will investigate what's required now...

**Comment by merks** [comment]
I opened this bug to track the work:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=581386

And there is this Gerrit:

https://git.eclipse.org/r/c/orbit/orbit-recipes/+/199491

But it already looks wrong.  All the stuff in the bin folder is not in the repo so I need to fix that.

**Comment by merks** [comment]
I fixed that problem.  I don't know the process for updating this file:

https://git.eclipse.org/r/c/orbit/orbit-recipes/+/199491/2/apache/ant/org.apache.ant_1.10.13/src/eclipse/ip_log.xml

**Comment by merks** [comment]
The build https://ci.eclipse.org/orbit/job/gerrit-orbit-recipes/2048 is failing with this:
```
08:10:42 [INFO] No MANIFEST.MF file found, generating manifest.
08:10:43 [WARNING] Manifest org.eclipse.orbit.bundles:org.apache.ant:eclipse-bundle-recipe:1.10.13-SNAPSHOT : Unused Export-Package instructions: [*.internal.*, *.impl.*] 
08:10:43 [ERROR] Manifest org.eclipse.orbit.bundles:org.apache.ant:eclipse-bundle-recipe:1.10.13-SNAPSHOT : The default package '.' is not permitted by the Import-Package syntax.
 This can be caused by compile errors in Eclipse because Eclipse creates
valid class files regardless of compile errors.
The following package(s) import from the default package null
08:10:43 [ERROR] Error(s) found in manifest configuration
```
So I think something needs to be fixed in the osgi.bnd, but that's a bit beyond my skill level.  :-(

I have about 1/2 hour left before I have to leave for most of the day...

**Comment by akurtakov** [comment]
@SarikaSinha Would you please help with that?

**Comment by SarikaSinha** [comment]
I will look at it.

**Comment by SarikaSinha** [comment]
Not able to figure out the error.
Roland had asked to create a new folder to support the new version and when it used to get stable, We used to remove the old folder.
@merks Have you downloaded the latest src and bin jar and taken the latest xsl files and jars from there?

**Comment by merks** [comment]
No, I copied the old folder to the new name and ran the script, assuming that's all I needed to do.  The documentation isn't so clear on exactly what one needs to do and were things come from. Maybe what I made is just a mess.  Sorry.

**Comment by SarikaSinha** [comment]
I will be able to work on this over the weekend for M3 if it doesn't go through now.

**Comment by jonahgraham** [comment]
I identified why the build is failing, reproduced from gerrit:

The reason the build is failing is because of a workaround for the security manager. 

There is a new class in ant-launcher.jar in the default package called allow (lowercase on purpose). The description of its purpose required reading the javadoc for the class at https://github.com/apache/ant/blob/014e94e5d1e289acaf8a4e6e7234662c065d48b2/src/main/allow.java#L30 in combination with JEP 411 at https://openjdk.org/jeps/411

(PS FWIW to identify the problem I did:
1- a binary search on all the dependencies in the pom.xml to find out which jar file caused EBR to complain. 
2- That identified that the build worked fine if I excluded ant-launcher.jar
3- I extracted the ant-launcher.jar and saw the unusual allow.class
4- Looked up allow.java in Ant on github)

The follow on thing I realized is that the contents of the bin and etc directory are not up to date. There is a README.adoc in the ant directory that explains this a little.

The [create-ant-recipes.sh](http://create-ant-recipes.sh/) is basically useless as what it generates is nowhere near what is used to make the ant pom.xml and other files. So I will delete it with my update if I can get this to work.

**Comment by jonahgraham** [comment]
> The follow on thing I realized is that the contents of the bin and etc directory are not up to date. 

I see that @SarikaSinha had identified this ^^^ already as an issue in https://github.com/eclipse-platform/eclipse.platform/issues/311#issuecomment-1398474882

> @merks Have you downloaded the latest src and bin jar and taken the latest xsl files and jars from there?

**Comment by merks** [comment]
I think we should take our time and do this properly.  And properly document what needs to be done.  I'm sure I didn't do anything quite right...  😢

**Comment by jonahgraham** [comment]
Yes. I am not going to be able to do it today regardless as this seems more complicated. The place to document this is apache/ant/README.adoc (I think) in the Orbit repo.

**Comment by SarikaSinha** [comment]
https://wiki.eclipse.org/Ant/How_to_add_new_Ant_versions

This is our original document for Updating Ant version, which needs to be updated for Orbit.

**Comment by SarikaSinha** [comment]
@jonahgraham 
Doing this first time after change in IP rules, as no IP approval is required now so do we still need ip_log.xml ?
As it has :
 <legal>
      <ipzilla bug_id="23806"/>
      <license>
        <name>Apache License, 2.0</name>
        <reference>http://www.apache.org/licenses/LICENSE-2.0.txt</reference>
      </license>
      <package>ant-1.10.12.jar</package>
    </legal>

**Comment by jonahgraham** [comment]
> Doing this first time after change in IP rules, as no IP approval is required now so do we still need ip_log.xml ?

We still need IP approval, just the process is much more streamlined. I don't think the tooling works for how ant is repackaged. 

However we can punt this down the road as 1.10.13 uses the same approvals as 1.10.12, so using CQ 23806 is the correct thing to do with `s/1.10.12/1.10.13/` in the ip_log

**Comment by akurtakov** [comment]
Closing in favor of https://github.com/eclipse-platform/eclipse.platform/issues/639
